extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -462,21 +462,21 @@ export async function makeHttpRequest<B>(
 		requestUrl = `${account.properties.providerSettings.settings.armResource.endpoint}${path}`;
 	}
 
-	let response: AzureNetworkResponse<HttpClientResponse<B>> | undefined = undefined;
+	let response: AzureNetworkResponse<B | ErrorResponseBodyWithError> | undefined = undefined;
```

Comment:
This also then needed to get fixed, since previously the error property was just stuffed on the HttpClientResponse. But that isn't really correct - there's really just two separate response types we can get, either the correct response (B) or an error. So having it typed like this makes it so the caller doesn't have to know anything about the error response type since we'll be either throwing an error or adding it to the error messages array if that happens.